tests/test_request_fingerprinter.py

```diff
@@ -0,0 +1,243 @@
+import sys
+from unittest.mock import patch
+
+import pytest
+from packaging.version import Version
+from scrapy import __version__ as SCRAPY_VERSION
+
+if Version(SCRAPY_VERSION) < Version("2.7"):
+    pytest.skip("Skipping tests for Scrapy < 2.7", allow_module_level=True)
+
+from importlib.metadata import version as package_version
+
+from scrapy import Request, Spider
+from web_poet import ItemPage, WebPage
+
+from scrapy_poet import ScrapyPoetRequestFingerprinter
+
+from . import get_crawler as _get_crawler
+
+ANDI_VERSION = Version(package_version("andi"))
+
+SETTINGS = {
+    "DOWNLOADER_MIDDLEWARES": {
+        "scrapy_poet.InjectionMiddleware": 543,
+    },
+    "REQUEST_FINGERPRINTER_CLASS": ScrapyPoetRequestFingerprinter,
+}
+
+
+def get_crawler(spider_cls=None, settings=None):
+    settings = SETTINGS if settings is None else settings
+    kwargs = {}
+    if spider_cls is not None:
+        kwargs["spider_cls"] = spider_cls
+    return _get_crawler(settings=settings, **kwargs)
+
+
+def test_no_deps_vs_dep():
+    class TestSpider(Spider):
+        name = "test_spider"
+
+        async def parse_page(self, response, page: WebPage):
+            pass
+
+    crawler = get_crawler(spider_cls=TestSpider)
+    fingerprinter = crawler.request_fingerprinter
+    request1 = Request("https://toscrape.com")
+    fingerprint1 = fingerprinter.fingerprint(request1)
+    request2 = Request("https://toscrape.com", callback=crawler.spider.parse_page)
+    fingerprint2 = fingerprinter.fingerprint(request2)
+    assert fingerprint1 != fingerprint2
+
+
+def test_same_deps():
```

Comment:
The spider is the same here with the one above. We can probably remove this and instead add a 3rd test above:

```python
    request3 = Request("https://toscrape.com", callback=crawler.spider.parse_page)
    fingerprint3 = fingerprinter.fingerprint(request3)
    assert fingerprint2 == fingerprint2
```

Also adding a comment to emphasize better that its testing for same deps.